javatests/arcs/core/analysis/AbstractSetTest.kt

```diff
@@ -0,0 +1,110 @@
+package arcs.core.analysis
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+
+/** Tests for BoundedAbstractValue. */
+@RunWith(JUnit4::class)
+class AbstractSetTest {
+
+    private val bottom = AbstractSet.getBottom<Int>()
+    private val top = AbstractSet.getTop<Int>()
+    private val odds = AbstractSet<Int>(setOf(1, 3, 5, 7, 9))
+    private val evens = AbstractSet<Int>(setOf(0, 2, 4, 6, 8))
+    private val multiplesOfThree = AbstractSet<Int>(setOf(3, 6, 9, 15))
+    private val naturals = AbstractSet<Int>(setOf(0, 1, 2, 3, 4, 5, 6, 7, 8, 9))
+
+    @Test
+    fun bottomConstruction() {
+        assertThat(bottom.isBottom).isTrue()
+        assertThat(bottom.isTop).isFalse()
+        assertThat(bottom.set).isNull()
+    }
+
+    @Test
+    fun topConstruction() {
+        assertThat(top.isBottom).isFalse()
+        assertThat(top.isTop).isTrue()
+        assertThat(top.set).isNull()
+    }
+
+    @Test
+    fun valueConstruction() {
+        assertThat(odds.isBottom).isFalse()
+        assertThat(odds.isTop).isFalse()
+        assertThat(requireNotNull(odds.set)).containsExactly(1, 3, 5, 7, 9)
+    }
+
+    @Test
+    fun prettyPrint() {
+        assertThat("$top").isEqualTo("TOP")
+        assertThat("$bottom").isEqualTo("BOTTOM")
+        assertThat("$odds").isEqualTo("[1, 3, 5, 7, 9]")
+    }
+
+    @Test
+    fun isEquivalentTo() {
+        with(bottom) {
```

Comment:
Can you split this test up into three separate tests? `isEquivalentTo_bottom`, `isEquivalentTo_top`, `isEquivalentTo_odds`?

Reply:
Done.